src/pu-command.c

```diff
@@ -270,6 +271,17 @@ pu_command_context_parse(PuCommandContext *context,
     if (context->command && context->command->option_entries != NULL) {
         g_option_context_add_main_entries(context->option_context,
                                           context->command->option_entries, NULL);
+    } else {
+        /* Main option entries should not contain the G_OPTION_REMAINING, as it
+         * would override the one provided with the command specific one. For
+         * that reason, add the main usage here. */
+        GOptionEntry extra_entries[] = {
+            { G_OPTION_REMAINING, 0, G_OPTION_FLAG_NONE, G_OPTION_ARG_STRING_ARRAY,
+                arg_remaining, NULL, "COMMAND" },
```

Comment:
If `arg_remaining` is `NULL`at this point, we get a null pointer dereference when `g_option_context_parse` tries to write to it while parsing.